### PR TITLE
Feat mimo audio

### PR DESCRIPTION
**Pull request by Mozoltov821**
Support MiMo-Audio-7B-Instruct model [sgl-project/sglang-jax#726](https://github.com/sgl-project/sglang-jax/issues/726)
Milestone 1: Implement Basic MiMo-Audio Model (Text-Only Output)

## Accuracy Tests

setup environment

```
uv sync --extra tpu,multimodal
```

launch_server command
```
uv run python3 -m sgl_jax.launch_server    --multimodal  --model-path XiaomiMiMo/MiMo-Audio-7B-Instruct  --trust-remote-code 
```

## Test Command

1. Online FIle
```
curl -X POST "http://localhost:30000/v1/audio/transcriptions" \
    -F "url=https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen3-Omni/cookbook/asr_en.wav" \
    -F "model=whisper-1" \
    -F "language=zh" \
    -F "response_format=json"
```

```
{"text":"Oh yeah yeah he wasn't even that big when i started listening to him but and his solo music didn't do overly well but he did very well when he started writing for other people","task":"transcribe","language":"zh","duration":null,"segments":null,"words":null,"usage":null}

```

2. Local File

[0_43fce45f-3809-48f8-b059-351072a4743c.wav](https://github.com/user-attachments/files/25513459/0_43fce45f-3809-48f8-b059-351072a4743c.wav)

```
curl -X POST "http://localhost:30000/v1/audio/transcriptions" -F "file=@0_43fce45f-3809-48f8-b059-351072a4743c.wav" -F "model=whisper-1" -F "language=zh" -F "response_format=json"
```

```
{"text":"北京今天晴转多云，气温在25到34摄氏度之间，南风3到4级，体感会比较闷热，建议你注意防暑降温，并尽量避开中午最热的时候出门。另外，
  外线也比较强，记得做好防晒措施。","task":"transcribe","language":"zh","duration":null,"segments":null,"words":null,"usage":null}
```

### CODE REVIEW

File: python/sgl_jax/srt/layers/attention/native_backend.py [hunk 1]
```diff
@@ -261,6 +261,7 @@ def forward_attention(
         attn_logits = _apply_decode_mask(attn_logits, seq_lengths)
 
     # Softmax
+    attn_logits = attn_logits - jnp.max(attn_logits, axis=-1, keepdims=True)
```
Comment (pengchengneo) [inline python/sgl_jax/srt/layers/attention/native_backend.py:264]: Can this modification be placed in the Mimo-Audio model implementation? Placing it here would affect all models using the native attention backend.

File: python/sgl_jax/srt/multimodal/configs/mimo_audio/__init__.py
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/configs/mimo_audio/__init__.py:1]: rename this folder to ../configs/mimo_audio/..

File: python/sgl_jax/srt/multimodal/manager/scheduler/audio_backbone_scheduler.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+        return actual_size
+
+    def event_loop_normal(self):
+        """Main blocking loop for processing requests.
+
+        Repeatedly polls the communication_backend for requests, processes them
+        through the backbone model, and sends results to the next stage.
+        """
+        # Set mesh context for all JAX operations (required for shard_map, KV cache updates, etc.)
+        with self.mesh:
```
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/manager/scheduler/audio_backbone_scheduler.py]: this maybe useless, refer to other scheduler ?

File: python/sgl_jax/srt/multimodal/manager/io_struct.py [hunk 3]
```diff
@@ -145,10 +188,31 @@ class TokenizedGenerateOmniReqInput:
...
+
+
 @dataclasses.dataclass
 class OmniInputs:
     audio_features: np.ndarray | None = None
     audio_feature_lengths: list[int] | None = None
+
+
+@dataclasses.dataclass
+class VLMMInputs:
```
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/manager/io_struct.py]: why modify this code of VLM ?

File: python/sgl_jax/srt/multimodal/manager/stage.py [hunk 4]
```diff
@@ -200,6 +221,14 @@ def get_scheduler_class(name: str):
         return VitScheduler
     elif name == "embedding":
         return EmbedScheduler
+    elif name == "audio":
```
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/manager/stage.py]: elif name in {"audio", "audio_encoder", "audio_decoder"}:

File: python/pyproject.toml [hunk 5]
```diff
@@ -43,7 +43,9 @@ dependencies = [
     "pathwaysutils",
     "omegaconf",
     "imageio[ffmpeg]",
+    "qwix",
     "librosa",
+    "soundfile",
```
Comment (pengchengneo) [inline python/pyproject.toml]: move to multimodal

File: python/pyproject.toml [hunk 6]
```diff
@@ -43,7 +43,9 @@ dependencies = [
     "pathwaysutils",
     "omegaconf",
     "imageio[ffmpeg]",
+    "qwix",
```
Comment (pengchengneo) [inline python/pyproject.toml]: move to multimodal

File: python/pyproject.toml
Comment (pengchengneo) [inline python/pyproject.toml:175]: add torchaudio to avoid import torch[GPU] by dependency

**Comment by pengchengneo** [comment]
@Mozoltov821  please rebase main

File: python/sgl_jax/srt/multimodal/manager/scheduler/audio_backbone_scheduler.py (hunk 2 above)
Comment (Mozoltov821) [reply python/sgl_jax/srt/multimodal/manager/scheduler/audio_backbone_scheduler.py]: solved

File: python/pyproject.toml (hunk 5 above)
Comment (Mozoltov821) [reply python/pyproject.toml]: solved

File: python/pyproject.toml (hunk 6 above)
Comment (Mozoltov821) [reply python/pyproject.toml]: solved

File: python/pyproject.toml
Comment (Mozoltov821) [reply python/pyproject.toml:175]: done

File: python/sgl_jax/srt/multimodal/manager/schedule_batch.py [hunk 7]
```diff
@@ -169,7 +183,43 @@ class Req:
...
+    text_logits: jax.Array | None = None
+
+    # Autoregressive generation state
+    is_prefill: bool = True  # True for first forward pass, False for decode steps
+    is_finished: bool = False  # True when EOS token is generated
+
     def to_stage_reqs(self, scheduler: str):
+        import logging
+        logger = logging.getLogger(__name__)
+        logger.info(
```
Comment (pengchengneo) [inline python/sgl_jax/srt/multimodal/manager/schedule_batch.py]: this should be debug log

File: python/sgl_jax/srt/multimodal/manager/schedule_batch.py [hunk 8]
```diff
@@ -252,3 +332,239 @@ def from_stage(stage_result: Any, req_store: dict):
...
+        The backbone expects input_ids with shape [B, 1+audio_channels, seq_len] where:
+        - Channel 0: text tokens (with -100 padding for non-first timesteps in each group)
+        - Channels 1-8: audio codes (or speech_empty_ids for text patches)
+
+        Returns:
+            input_ids: [1, 9, seq_len] tensor ready for backbone forward pass
+        """
+        import logging
+
+        logger = logging.getLogger(__name__)
```
Comment (Mozoltov821) [inline python/sgl_jax/srt/multimodal/manager/schedule_batch.py]: removed

**Review by pathfinder-pf — APPROVED** [review]
